### ISSUE TITLE
Compilation error when using grin_wallet_libwallet as dependency

### ISSUE THREAD

**Issue by jafalter** [mimblewimble/grin-wallet#556]
**Describe the bug**
When adding grin_wallet_libwallet as a dependency to a Rust project like this:
![image](https://user-images.githubusercontent.com/7274756/103210220-600c1200-4905-11eb-8880-041b52346f05.png)`
I receive a compilation error which seems to be coming from an issue with a dalek and the grin-core dependency. The bug appears for any version of grin_wallet_libwallet above 3.0.0. (2.1.0 and below works fine)

```
error: aborting due to 26 previous errors

Some errors have detailed explanations: E0026, E0027, E0308, E0560, E0599, E0609, E0615.
For more information about an error, try `rustc --explain E0026`.
error: could not compile `grin_wallet_libwallet`.
```

**To Reproduce**
1. `mkdir minimal`
2. `cd minimal`
3. `cargo init`
4. Add `grin_wallet_libwallet = "4.0.0"` as dependency to `Cargo.toml`
5. `cargo build`

**Expected behavior**
Should compile without errors

**Desktop (please complete the following information):**
 - Ubuntu 20.04
 - Rust 1.47.0

**Screenshots**
![image](https://user-images.githubusercontent.com/7274756/103210692-b463c180-4906-11eb-82bb-fd33d43ad0bc.png)
![image](https://user-images.githubusercontent.com/7274756/103210696-bcbbfc80-4906-11eb-9c36-075c0ae8df22.png)
![image](https://user-images.githubusercontent.com/7274756/103210703-c47ba100-4906-11eb-9211-f848d7b8043d.png)
![image](https://user-images.githubusercontent.com/7274756/103210715-ccd3dc00-4906-11eb-959d-7c76891e71ae.png)
![image](https://user-images.githubusercontent.com/7274756/103210724-d5c4ad80-4906-11eb-8cdf-3f19f398ce53.png)
![image](https://user-images.githubusercontent.com/7274756/103210748-dd845200-4906-11eb-9226-d8eb589f0465.png)
![image](https://user-images.githubusercontent.com/7274756/103210760-e7a65080-4906-11eb-983b-3b3006303d47.png)

**Comment by trevyn** [comment]
This appears to be fixed in `grin_wallet_libwallet = 5.0.1`.

I can reproduce the issue with Rust 1.50 and `grin_wallet_libwallet = 4.0.0`, but this works for me:

```toml
[dependencies]
grin_wallet_libwallet = "5.0.1"
serde = "=1.0.118"
```

(Note that `serde = "=1.0.118"` is required due to https://github.com/serde-rs/serde/issues/1949. This is fixed in https://github.com/mimblewimble/grin/pull/3585, but that is not yet released to crates.io.)

**Comment by quentinlesceller** [comment]
Can confirm @trevyn message. Thanks! Closing.
